src/main/java/org/openjdk/jextract/impl/OutputFactory.java

```diff
@@ -232,50 +168,36 @@ public Void visitFunction(Declaration.Function funcTree, Declaration parent) {
         if (Skip.isPresent(funcTree)) {
             return null;
         }
-        //generate static wrapper for function
-        String unsupportedType = UnsupportedLayouts.firstUnsupportedType(funcTree.type());
-        if (unsupportedType != null) {
-            warn("skipping " + funcTree.name() + " because of unsupported type usage: " +
-                    unsupportedType);
-            return null;
-        }
-
-        FunctionDescriptor descriptor = Type.descriptorFor(funcTree.type()).orElse(null);
-        if (descriptor == null) {
-            return null;
-        }
 
         // check for function pointer type arguments
-        int i = 0;
         for (Declaration.Variable param : funcTree.parameters()) {
             Type.Function f = Utils.getAsFunctionPointer(param.type());
             if (f != null) {
-                if (! generateFunctionalInterface(param, f)) {
+                if (! generateFunctionalInterface(JavaFunctionalInterfaceName.getOrThrow(param), f)) {
```

Comment:
Pre-existing, but this looks potentially problematic: it seems like it is possible that we generate a FI class for one of the parameters, then find that a following parameter is Skipped, and drop the function. Leaving a dangling FI class for the other param (similar if the return type is skipped).

These FI classes are unique to the particular function AFAICS. See also my related comment in UnsupportedFilter. I think this part can be simplified more by putting the Skip on the function tree instead. Then we would never see an unsupported/Skipped type here.